packages/ui5-proxy-middleware/test/base/proxy.test.ts

```diff
@@ -0,0 +1,100 @@
+import { ui5Proxy } from '../../src';
+import * as hpm from 'http-proxy-middleware';
+import * as utils from '../../src/base/utils';
+import { ToolsLogger } from '@sap-ux/logger';
+
+describe('ui5Proxy', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    test('ui5Proxy: default params', () => {
```

Comment:
Can we refactor this test (and the next one)? I have a few points to note about it:
* it's actually more complicated than the function `ui5Proxy()` that it's testing. Tests need to simple and dumb
* it doesn't seem to have a clear purpose - we need to be able to succinctly explain what a test is doing and I'm not able to answer that here
* it's testing more than one thing - ideally if a test breaks it needs to lead back to one very clear area in the code. This test is checking:
  - `createProxyMiddleware` is called with correct params
  - If a default filter function is a function, encoding is `undefined`
  - if `onProxyReq` is returned it's a function and so on

Reply:
OK, will refactor those

Reply:
Done